plugins/inputs/docker/docker.go

```diff
@@ -134,6 +148,15 @@ func (d *Docker) Gather(acc telegraf.Accumulator) error {
 		d.client = c
 	}
 
+	// Create label filters
+	if len(d.LabelInclude) != 0 {
```

Comment:
I'm not sure I completely understand the question.  But yes, every time Gather runs, it would compile this filter.